File: android-ui/settings/src/androidMain/kotlin/io/github/viacheslav/chugunov/moniqo/ui/settings/screen/SettingsIntent.kt
package io.github.viacheslav.chugunov.moniqo.ui.settings.screen

import io.github.viacheslav.chugunov.moniqo.core.model.AppLanguage
import io.github.viacheslav.chugunov.moniqo.core.model.AppTheme

internal sealed interface SettingsIntent {
    data class ChangeTheme(val theme: AppTheme) : SettingsIntent

    data class ChangeDealRanges(val goodMax: Float, val averageMax: Float) : SettingsIntent

    data object ResetRanges : SettingsIntent

    data object OpenRangeEditor : SettingsIntent

    data object CloseRangeEditor : SettingsIntent

    data class ChangeLanguage(val language: AppLanguage) : SettingsIntent

    data object OpenLanguagePicker : SettingsIntent

    data object CloseLanguagePicker : SettingsIntent
}

File: iosApp/iosApp/settings/screen/SettingsIntent.swift
import Foundation

enum SettingsIntent {
    case changeTheme(AppThemeSetting)
    case changeLanguage(AppLanguageSetting)
    case changeDealRanges(good: Int, medium: Int)
    case resetRanges
    case openRangeEditor
    case closeRangeEditor
    case openLanguagePicker
    case closeLanguagePicker
}
